### ISSUE TITLE
Support different dimensions

### ISSUE THREAD

**Issue by dryman** [una-dinosauria/Rayuela.jl#39]
Does this package support other dimensions than 128?
We'd like to test it on GloVe, which is 100.

**Comment by una-dinosauria** [comment]
Yep, just add your data reading functions to https://github.com/una-dinosauria/Rayuela.jl/blob/ccf22abef46d29651e4115ffa47b648229ae8571/src/read_datasets.jl#L4

**Comment by dryman** [comment]
Does the following code in cuda matter?
```
deps/src/cudautils.cu
93:  // FIXME hard-coding 256 entries in each codebook, and 128 dimensions
95:  const int D = 128; // dimensionality of each vector
```

**Comment by una-dinosauria** [comment]
Oh right, sorry. The GPU part is hardcoded.

**Comment by una-dinosauria** [comment]
I checked again, since I definitely ran this code on the GPU for my thesis.

The function `veccost` that you linked above is not called from the Julia part. `veccost2` is
https://github.com/una-dinosauria/Rayuela.jl/blob/ccf22abef46d29651e4115ffa47b648229ae8571/deps/src/cudautils.cu#L154-L155

which only hardcodes the number of entries in the codebook (not the dataset dimensionality). I'm changing the name of this issue to reflect that.

**Comment by una-dinosauria** [comment]
Also opened #41 to keep track of code deletion.
